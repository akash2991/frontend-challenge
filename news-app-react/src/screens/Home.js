import React, { useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import LinearProgress from '@material-ui/core/LinearProgress';
import { MainBar, NewsArticleList, Picker } from '../components';
import { useFetchArticles } from '../service';
import { pickerDataCategory, pickerDataCountry, useDebounce } from '../utils';

const useStyles = makeStyles((theme) => ({
  cardContainer: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-between',
  },
  mainContainer: {
    width: '100%',
    // backgroundColor: 'rgb(211,211,211)',
  },
  pickerContainer: {
    display: 'flex',
    marginTop: 10,
  },
}));

export default function Home() {
  const classes = useStyles();
  const [searchQuery, setSearchQuery] = useState('');
  const [country, setCountry] = useState('');
  const [category, setCategory] = useState('');
  const debouncedSearchQuery = useDebounce(searchQuery, 500);
  const { data = [], isLoading, error, hideArticle } = useFetchArticles({
    searchQuery: debouncedSearchQuery,
    country,
    category,
  });

  function handleSearchQuery(e) {
    setSearchQuery(e.target.value);
  }

  return (
    <div className={classes.mainContainer}>
      <MainBar handleSearchFieldInput={handleSearchQuery} />

      {isLoading && <LinearProgress color="secondary" />}

      <div className={classes.pickerContainer}>
        <Picker
          pickerData={pickerDataCategory}
          label="Category"
          getPickerValue={setCategory}
        />
        <Picker
          pickerData={pickerDataCountry}
          label="Country"
          getPickerValue={setCountry}
        />
      </div>

      <NewsArticleList data={data} hideArticle={hideArticle} />
    </div>
  );
}
