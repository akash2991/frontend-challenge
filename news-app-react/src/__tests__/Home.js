import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../screens/Home';

it('renders correctly', () => {
  const tree = renderer.create(<Home></Home>).toJSON();
  expect(tree).toMatchSnapshot();
});
