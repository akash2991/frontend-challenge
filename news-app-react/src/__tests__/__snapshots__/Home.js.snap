// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="makeStyles-mainContainer-2"
  onKeyDown={[Function]}
  onKeyPress={[Function]}
>
  <div
    className="makeStyles-grow-4"
  >
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="open drawer"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-5 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
        <h6
          className="MuiTypography-root makeStyles-title-6 MuiTypography-h6 MuiTypography-noWrap"
        >
          News App
        </h6>
        <div
          className="makeStyles-search-7"
        >
          <div
            className="makeStyles-searchIcon-8"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <div
            className="MuiInputBase-root makeStyles-inputRoot-9"
            onClick={[Function]}
          >
            <input
              aria-label="search"
              className="MuiInputBase-input makeStyles-inputInput-10"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Filter news by keyword"
              type="text"
            />
          </div>
        </div>
      </div>
    </header>
  </div>
  <div
    className="MuiLinearProgress-root MuiLinearProgress-colorSecondary MuiLinearProgress-indeterminate"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-bar MuiLinearProgress-barColorSecondary MuiLinearProgress-bar1Indeterminate"
      style={Object {}}
    />
    <div
      className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorSecondary"
      style={Object {}}
    />
  </div>
  <div
    className="makeStyles-pickerContainer-3"
  >
    <div
      className="MuiFormControl-root makeStyles-formControl-11"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink={false}
        id="simple-select-outlined-Category"
      >
        Category
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="simple-select-outlined-Category"
          className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </div>
        <input
          aria-hidden={true}
          className="MuiSelect-nativeInput"
          onAnimationStart={[Function]}
          onChange={[Function]}
          required={false}
          tabIndex={-1}
          value=""
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-13 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-15"
          >
            <span>
              Category
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="MuiFormControl-root makeStyles-formControl-11"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink={false}
        id="simple-select-outlined-Country"
      >
        Country
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="simple-select-outlined-Country"
          className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </div>
        <input
          aria-hidden={true}
          className="MuiSelect-nativeInput"
          onAnimationStart={[Function]}
          onChange={[Function]}
          required={false}
          tabIndex={-1}
          value=""
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-13 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-15"
          >
            <span>
              Country
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    className="makeStyles-cardContainer-17"
  >
    <h6
      className="MuiTypography-root makeStyles-title-6 MuiTypography-h6 MuiTypography-noWrap"
    >
      News is not available. Please refresh the page
    </h6>
  </div>
</div>
`;
